spec/lib/category_guardian_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+RSpec.describe CategoryGuardian do
+  fab!(:admin) { Fabricate(:admin) }
+  fab!(:user) { Fabricate(:user) }
+  fab!(:can_create_user) { Fabricate(:user) }
+  fab!(:category) { Fabricate(:category) }
+
+  describe "can_post_in_category?" do
+    it 'returns true for admin and regular user when not restricted category' do
+      expect(Guardian.new.can_post_in_category?(category)).to eq(false)
+      expect(Guardian.new(admin).can_post_in_category?(category)).to eq(true)
+      expect(Guardian.new(user).can_post_in_category?(category)).to eq(true)
+    end
+
+    it 'returns true for admin and memebers of priviliged groups for restricted category' do
```

Comment:
I think it would be really be nicer to have a `context "when ..."` for each permission_type state